execute/executor.go

```diff
@@ -266,10 +264,15 @@ func (es *executionState) do(ctx context.Context) {
 					default:
 						err = fmt.Errorf("%v", e)
 					}
-					es.abort(fmt.Errorf("panic: %v\n%s", err, debug.Stack()))
-					if entry := es.logger.Check(zapcore.InfoLevel, "Execute source panic"); entry != nil {
-						entry.Stack = string(debug.Stack())
-						entry.Write(zap.Error(err))
+
+					if _, ok := err.(memory.LimitExceededError); ok {
+						es.abort(err)
```

Comment:
Instead of the hanging else, can you do a `return` and then remove the else part?